src/chat/chat_panel.rs

```diff
@@ -429,6 +429,9 @@ pub struct ChatPanel {
 
     #[rust]
     state: State,
+
+    #[rust]
+    portal_list_end_reached: bool,
```

Comment:
I will leave this separated as it's just to support the portal list workaround here.